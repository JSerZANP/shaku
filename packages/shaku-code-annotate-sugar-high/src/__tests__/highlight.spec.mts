import { expect, test } from "vitest";
import { highlight } from "../index.mjs";

const snippets = [
  `
// @highlight
function ChatRoom({ roomId }) {
         --------
  const [serverUrl, setServerUrl] = useState('https://localhost:1234');
  //      ^
  // [Hello]
  
  // @diff + start
  // Welcome
  // To
  // JSer!
  // @diff - start

  // @diff -
  console.log('hello')

  // @dim
  useEffect(() => {
     // (                                                     )
        const connection = createConnection(serverUrl, roomId);
     // (1                )
        connection.connect();
    
  	return () => {
           //(2                    )
             connection.disconnect();
  	};
   //(3                 )
  }, [serverUrl, roomId]);
}
`,
  `
function hello() {
  //        ^!
  //    [Hello world!]!
  //        ^
  //    [Shaku Shaku]
  const blog = "https://jser.dev"
}
  `,
  `
  function hello() {
    return <div
    //       ^
    // [Hello!]
    >
    Hello world!
  {/*    ^     */}
  {/* [Hello!] */}
  </div>
  }
    `,
];

test("highlight()", async () => {
  snippets.forEach((code) => {
    const html = highlight(code);
    expect(html).toMatchSnapshot();
  });
});
