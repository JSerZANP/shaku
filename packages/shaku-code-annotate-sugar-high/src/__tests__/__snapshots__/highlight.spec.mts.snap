// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`highlight() 1`] = `
"<pre class=\\"shaku\\"><code><div class=\\"sh__line\\"><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line highlight\\"><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">function</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--class\\" style=\\"color: var(--sh-class)\\">ChatRoom</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">{</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">roomId</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">}</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">{</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">         </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">-</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">-</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">-</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">-</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">-</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">-</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">-</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">-</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">const</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">[</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">serverUrl</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">,</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">setServerUrl</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">]</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">=</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">useState</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">&#039;</span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">https://localhost:1234</span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">&#039;</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">;</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"shaku-callout\\" style=\\"left:5ch\\"><span class=\\"shaku-callout-arrow\\" style=\\"left:5ch\\"></span>Hello</div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-insert\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--comment\\" style=\\"color: var(--sh-comment)\\">// Welcome
</span></div><div class=\\"sh__line diff diff-insert\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--comment\\" style=\\"color: var(--sh-comment)\\">// To
</span></div><div class=\\"sh__line diff diff-insert\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--comment\\" style=\\"color: var(--sh-comment)\\">// JSer!
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">console</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">.</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">log</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">&#039;</span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">hello</span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">&#039;</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line dim diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">useEffect</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">=</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&gt;</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">{</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">        </span><mark class=\\"shaku-inline-highlight\\" ><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">const</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">connection</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">=</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">createConnection</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">serverUrl</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">,</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">roomId</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">;</span></mark><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">        </span><mark class=\\"shaku-inline-highlight\\" data-id=\\"1\\"><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">connection</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">.</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">connect</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span></mark><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">;</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">    </span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  	</span><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">return</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">=</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&gt;</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">{</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">             </span><mark class=\\"shaku-inline-highlight\\" data-id=\\"2\\"><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">connection</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">.</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">disconnect</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span></mark><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">;</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  	</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">}</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">;</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">}</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">,</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><mark class=\\"shaku-inline-highlight\\" data-id=\\"3\\"><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">[</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">serverUrl</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">,</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">roomId</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">]</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span></mark><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">;</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line diff diff-delete\\"><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">}</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div></code></pre>"
`;

exports[`highlight() 2`] = `
"<pre class=\\"shaku\\"><code><div class=\\"sh__line\\"><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">function</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">hello</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">{</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--comment\\" style=\\"color: var(--sh-comment)\\">//        ^
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--comment\\" style=\\"color: var(--sh-comment)\\">//    [Hello world!]
</span></div><div class=\\"shaku-callout\\" style=\\"left:8ch\\"><span class=\\"shaku-callout-arrow\\" style=\\"left:4ch\\"></span>Shaku Shaku</div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">const</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">blog</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">=</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">&quot;</span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">https://jser.dev</span><span class=\\"sh__token--string\\" style=\\"color: var(--sh-string)\\">&quot;</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">}</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span></div></code></pre>"
`;

exports[`highlight() 3`] = `
"<pre class=\\"shaku\\"><code><div class=\\"sh__line\\"><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">function</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">hello</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">{</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">    </span><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">return</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&lt;</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">div</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"shaku-callout\\" style=\\"left:7ch\\"><span class=\\"shaku-callout-arrow\\" style=\\"left:6ch\\"></span>Hello!</div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">    </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&gt;</span><span class=\\"sh__token--jsxliterals\\" style=\\"color: var(--sh-jsxliterals)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--jsxliterals\\" style=\\"color: var(--sh-jsxliterals)\\">    Hello world!</span><span class=\\"sh__token--jsxliterals\\" style=\\"color: var(--sh-jsxliterals)\\">
</span></div><div class=\\"shaku-callout\\" style=\\"left:8ch\\"><span class=\\"shaku-callout-arrow\\" style=\\"left:3ch\\"></span>Hello!</div><div class=\\"sh__line\\"><span class=\\"sh__token--jsxliterals\\" style=\\"color: var(--sh-jsxliterals)\\">  </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&lt;/</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">div</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&gt;</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">}</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">    </span></div></code></pre>"
`;

exports[`highlight() 4`] = `
"<pre class=\\"shaku\\"><code><div class=\\"sh__line\\"><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line highlight abc e123\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">function</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">hello</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">(</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">)</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">{</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">    </span><span class=\\"sh__token--keyword\\" style=\\"color: var(--sh-keyword)\\">return</span><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\"> </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&lt;</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">div</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&gt;</span><span class=\\"sh__token--jsxliterals\\" style=\\"color: var(--sh-jsxliterals)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--jsxliterals\\" style=\\"color: var(--sh-jsxliterals)\\">    Hello world!</span><span class=\\"sh__token--jsxliterals\\" style=\\"color: var(--sh-jsxliterals)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--jsxliterals\\" style=\\"color: var(--sh-jsxliterals)\\">  </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&lt;/</span><span class=\\"sh__token--identifier\\" style=\\"color: var(--sh-identifier)\\">div</span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">&gt;</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">  </span><span class=\\"sh__token--sign\\" style=\\"color: var(--sh-sign)\\">}</span><span class=\\"sh__token--break\\" style=\\"color: var(--sh-break)\\">
</span></div><div class=\\"sh__line\\"><span class=\\"sh__token--space\\" style=\\"color: var(--sh-space)\\">    </span></div></code></pre>"
`;
